x-pack/plugins/fleet/server/services/epm/elasticsearch/template/template.ts

```diff
@@ -102,7 +103,9 @@ export function getTemplate({
     isIndexModeTimeSeries,
   });
   if (template.template.settings.index.final_pipeline) {
-    throw new Error(`Error template for ${templateIndexPattern} contains a final_pipeline`);
+    throw new PackageESError(
+      `Error template for ${templateIndexPattern} contains a final_pipeline`
+    );
```

Comment:
I'm just not sure if these errors should be a 500. I found a few that happen when dealing with ES but I'm not totally sure what the correct response should be.

Reply:
I think you're right, that is indeed the template obtained from the package. I'll change it to `PackageInvalidArchiveError`